apps/examples/testcase/le_tc/kernel/tc_sched.c

```diff
@@ -673,6 +673,33 @@ static void tc_sched_task_setcanceltype(void)
 #endif
 #endif
 
+
+/**
+* @fn                   :tc_sched_set_get_affinity
+* @brief                :set and get cpu affinity for a task
+* @scenario             :check if the cpu affinity is properly set
+* API's covered         :sched_setaffinity, sched_getaffinity
+* Preconditions         :none
+* Postconditions        :none
+* @return               :void
+*/
+
+static void tc_sched_set_get_affinity(void)
+{
+	cpu_set_t affinity = 1 << 0;	/* Set affinity to core zero */
+#ifdef CONFIG_SMP
+	TC_ASSERT_EQ("sched_setaffinity", sched_setaffinity(0, sizeof(cpu_set_t), &affinity), OK);
+	affinity = 0;
+	TC_ASSERT_EQ("sched_getaffinity", sched_getaffinity(0, sizeof(cpu_set_t), &affinity), OK);
+	TC_ASSERT_EQ("sched_getaffinity", affinity, (1 << 0));
+#else
+	TC_ASSERT_EQ("sched_setaffinity", sched_setaffinity(0, sizeof(cpu_set_t), &affinity), -EINVAL);
```

Comment:
I think the goal is to use same code for both single core and multi core case. If so, this will not work. For example if the below code exists in some thread, it will only move forward in multi core case.

```
.....
if (sched_setaffinity(0, sizeof(cpu_set_t), &affinity) != OK) {
        printf("setaffinity failed, exit\n");
        exit();
} 
/* do work related to thread */
.....
```
and it fails for non smp case always.

Reply:
Yes. You are right. It will fail and return error. 
This is what @sunghan-chang  asked me to do.

Reply:
As per discussion, I modified to ENOTSUP